### PR TITLE
F datasource rest api

### PR DESCRIPTION
**Pull request by ch18042690795**
F datasource rest api

### CODE REVIEW

File: starfish-api/src/main/java/org/metahut/starfish/api/controller/DatasourceController.java [hunk 1]
```diff
@@ -1,19 +1,58 @@
...
     /**
      * datasource test connection
+     *
      * @param type
      * @param parameter
      * @return
      */
     @GetMapping("testConnection")
-    ResultEntity testConnection(String type, String parameter);
+    @ApiOperation(value = "testConnection", notes = "test the datasource connection")
```
Comment (zhuangchong) [inline starfish-api/src/main/java/org/metahut/starfish/api/controller/DatasourceController.java:29]: `notes` first refer to the following writing method, which is convenient for internationalization later
```
notes="TEST_CONNECTION_DATASOURCE_NOTES"
```

File: starfish-api/src/main/java/org/metahut/starfish/api/dto/DatasourceDataRequestDTO.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package org.metahut.starfish.api.dto;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+@ApiModel(description  = "数据源请求类")
```
Comment (zhuangchong) [inline starfish-api/src/main/java/org/metahut/starfish/api/dto/DatasourceDataRequestDTO.java:6]: change to english.

File: starfish-api/src/main/java/org/metahut/starfish/api/enums/Status.java [hunk 3]
```diff
@@ -3,7 +3,8 @@
 public enum Status {
 
     SUCCESS(200, "success"),
-    UNKNOWN_EXCEPTION(10000, "UNKNOWN_EXCEPTION")
+    UNKNOWN_EXCEPTION(10000, "UNKNOWN_EXCEPTION"),
+    DATASOURCE_TEST_FAIL(20001, "datasource connection is fail")
```
Comment (zhuangchong) [inline starfish-api/src/main/java/org/metahut/starfish/api/enums/Status.java:7]: This spelling is wrong, add it in zh_CN and en_US under the i8n folder under the resources file.

File: starfish-datasource/starfish-datasource-pulsar/HELP.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Getting Started
```
Comment (zhuangchong) [inline starfish-datasource/starfish-datasource-pulsar/HELP.md:1]: The file name is usually called README.md

File: starfish-server/pom.xml [hunk 5]
```diff
@@ -41,6 +41,76 @@
...
+                    <groupId>javax.servlet</groupId>
+                    <artifactId>servlet-api</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-tomcat</artifactId>
+        </dependency>
+        <dependency>
```
Comment (zhuangchong) [inline starfish-server/pom.xml:96]: `hive jdbc` has been introduced under the hivedatasource module, why does it need to be introduced again here

File: starfish-server/pom.xml [hunk 6]
```diff
@@ -41,6 +41,76 @@
...
+                </exclusion>
+                <exclusion>
+                    <groupId>javax.servlet</groupId>
+                    <artifactId>servlet-api</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-tomcat</artifactId>
```
Comment (zhuangchong) [inline starfish-server/pom.xml:94]: No need to introduce tomcat

File: starfish-server/src/main/java/org/metahut/starfish/ServerApplication.java [hunk 7]
```diff
@@ -2,13 +2,14 @@
...
+import org.springframework.boot.web.servlet.support.SpringBootServletInitializer;
 import org.springframework.data.jpa.repository.config.EnableJpaAuditing;
 import org.springframework.scheduling.annotation.EnableAsync;
 
 @EnableAsync
-@SpringBootApplication
 @EnableJpaAuditing
-public class ServerApplication {
+@SpringBootApplication
+public class ServerApplication extends SpringBootServletInitializer {
```
Comment (zhuangchong) [inline starfish-server/src/main/java/org/metahut/starfish/ServerApplication.java:12]: Inheriting the SpringBootServletInitializer class is what needs to be done?

File: starfish-server/src/main/java/org/metahut/starfish/server/controller/DatasourceControllerImpl.java [hunk 8]
```diff
@@ -1,12 +1,16 @@
...
+import org.metahut.starfish.api.dto.DatasourceDataRequestDTO;
 import org.metahut.starfish.api.dto.ResultEntity;
+import org.metahut.starfish.api.enums.Status;
+import org.metahut.starfish.datasource.api.DatasourceResult;
 import org.metahut.starfish.server.service.DatasourceService;
 
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
+//@Api(tags = "数据源管理类")
```
Comment (zhuangchong) [inline starfish-server/src/main/java/org/metahut/starfish/server/controller/DatasourceControllerImpl.java:13]: Comments please clean up

**Review by zhuangchong — CHANGES_REQUESTED** [review]

File: starfish-server/pom.xml (hunk 5 above)
Comment (ch18042690795) [reply starfish-server/pom.xml:96]: hive jdbc not exist in hivedatasource

File: starfish-server/pom.xml (hunk 6 above)
Comment (ch18042690795) [reply starfish-server/pom.xml:94]: hive-standalone-metastore has a same tomcat jar ,introduce tomcat avoid to have confilcts

File: starfish-server/pom.xml [hunk 9]
```diff
@@ -68,12 +116,6 @@
         <dependency>
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-web</artifactId>
-            <exclusions>
-                <exclusion>
-                    <groupId>org.springframework.boot</groupId>
-                    <artifactId>spring-boot-starter-tomcat</artifactId>
-                </exclusion>
-            </exclusions>
```
Comment (zhuangchong) [inline starfish-server/pom.xml:76]: No need to delete here

File: starfish-api/src/main/java/org/metahut/starfish/api/enums/Status.java [hunk 10]
```diff
@@ -3,7 +3,8 @@
 public enum Status {
 
     SUCCESS(200, "success"),
-    UNKNOWN_EXCEPTION(10000, "UNKNOWN_EXCEPTION")
+    UNKNOWN_EXCEPTION(10000, "UNKNOWN_EXCEPTION"),
+    DATASOURCE_TEST_FAIL(20001, "datasource connection is fail:{0}")
```
Comment (zhuangchong) [inline starfish-api/src/main/java/org/metahut/starfish/api/enums/Status.java:7]: DATASOURCE_TEST_FAIL(20001, "DATASOURCE_TEST_FAIL")

DATASOURCE_TEST_FAIL is the key in the message.properties file

**Review by zhuangchong — CHANGES_REQUESTED** [review]

**Review by zhuangchong — APPROVED** [review]

**Pull request merged**